commit be0ef2054b305f43fa11202c8eb806764daac11d
Author: popcornmix <popcornmix@gmail.com>
Date:   Wed Mar 12 11:46:34 2014 +0000

    ASoc: Don't report S24_LE support, it produces white noise with xbmc

diff --git a/sound/soc/bcm/bcm2708-i2s.c b/sound/soc/bcm/bcm2708-i2s.c
index a179216..b25e158 100644
--- a/sound/soc/bcm/bcm2708-i2s.c
+++ b/sound/soc/bcm/bcm2708-i2s.c
@@ -718,7 +718,7 @@ static struct snd_soc_dai_driver bcm2708_i2s_dai = {
 		.channels_max = 2,
 		.rates =	SNDRV_PCM_RATE_8000_192000,
 		.formats =	SNDRV_PCM_FMTBIT_S16_LE
-				| SNDRV_PCM_FMTBIT_S24_LE
+				// | SNDRV_PCM_FMTBIT_S24_LE : disable for now, it causes white noise with xbmc
 				| SNDRV_PCM_FMTBIT_S32_LE
 		},
 	.capture = {
diff --git a/sound/soc/codecs/pcm5102a.c b/sound/soc/codecs/pcm5102a.c
index 126f1e9..7812d34 100644
--- a/sound/soc/codecs/pcm5102a.c
+++ b/sound/soc/codecs/pcm5102a.c
@@ -28,7 +28,7 @@ static struct snd_soc_dai_driver pcm5102a_dai = {
 		.channels_max = 2,
 		.rates = SNDRV_PCM_RATE_8000_192000,
 		.formats = SNDRV_PCM_FMTBIT_S16_LE |
-			   SNDRV_PCM_FMTBIT_S24_LE |
+			   // SNDRV_PCM_FMTBIT_S24_LE | : disable for now, it causes white noise with xbmc
 			   SNDRV_PCM_FMTBIT_S32_LE
 	},
 };
